ml-proto/src/check.ml

```diff
@@ -241,6 +239,13 @@ and check_arm c t ts arm =
   check_literal c [t] l;
   check_expr c (if fallthru then [] else ts) e
 
+and check_memop memop at =
+  require (memop.align = 1 || memop.align = 2 ||
+           memop.align = 4 || memop.align = 8)
```

Comment:
Agreed.

Extra advantage: subsequent indexed memory accesses can also key off the first access' alignment (base has an alignment, other indices don't because the alignment is implied).